journeyapps-evaluator/src/token-expressions/function/LegacyFunctionTokenExpression.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Legacy function token expression.
+ */
+import { TokenExpression, TokenExpressionOptions } from '../TokenExpression';
+import { ConstantTokenExpression } from '../constant/ConstantTokenExpression';
+import { FormatStringScope } from '../../definitions/FormatStringScope';
+
+export class LegacyFunctionTokenExpression extends TokenExpression {
+  static readonly TYPE = 'legacy-function-expression';
+  constructor(options: TokenExpressionOptions) {
+    super(LegacyFunctionTokenExpression.TYPE, { ...options, isFunction: true });
+  }
+
+  /**
+   * Generate a constant token expression from legacy function token expression.
+   * @param {boolean} [includeEscapeTags] if "{" and "}" format string escape tags should be included or not
+   */
+  toConstant(includeEscapeTags: boolean = false): ConstantTokenExpression {
+    let constantExpression = this.expression;
+    if (includeEscapeTags) {
+      constantExpression = '{' + constantExpression + '}';
+    }
+    return new ConstantTokenExpression({ expression: constantExpression, start: this.start });
+  }
+
+  tokenEvaluatePromise(scope: FormatStringScope): Promise<string> {
```

Comment:
Will this be implemented in the future or is it not necessary for FunctionTokenExpressions?

Reply:
`LegacyFunctionTokenExpression` does not use `$:` notation, e.g `on-press="goSomewhere()"`.
Without `$:` it can not be used in a `FormatString`

Example
```
<info value="Hello {getUserName()}"/>
```

This will fail to parse because showing 
> 'getUserName()' is not defined